### PR TITLE
enhancement: filter out data directory schema while checking the database permission

### LINKED ISSUE #2048: [Bug]: There is no need to restrict users’ permissions to access to schemas like  sys, information_schema
### ODC version

4.2.3

### OB version

all

### What happened?

If the sys schema is not added to the current project, you need to add it to the project to access it.
image

### What did you expect to happen?

There is no need to restrict users’ permissions to access schema like sys、 information_schema and mysql.

### How can we reproduce it (as minimally and precisely as possible)?

like above

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by MarkPotato777**
#### What type of PR is this?

type-enhancement

#### What this PR does / why we need it:
When users execute SQLs, ODC will check if they have the associated database permissions. However, some data directory schema is not filtered out which will cause a bad user experience, as these data directory schemas are default granted to every database user.
Therefore, in this PR, I filter out the `SYS` in the Oracle dialect type and `information_schema` in the MySQL dialect type no matter if these two schemas are placed in the project.

#### Which issue(s) this PR fixes:

Fixes #2048 

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/PreCheckRuntimeFlowableTask.java [hunk 1]
```diff
@@ -299,7 +299,8 @@ private void doSqlCheckAndDatabasePermissionCheck(TaskEntity preCheckTaskEntity,
                 }
             }
             unauthorizedDatabases =
-                    databaseService.filterUnauthorizedDatabases(schemaName2PermissionTypes, connectionConfig.getId());
+                    databaseService.filterUnauthorizedDatabases(schemaName2PermissionTypes, connectionConfig.getId(),
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/PreCheckRuntimeFlowableTask.java:302]: There is also a class named PreCheckRuntimeFlowableTaskCopied. Does it also need to be modified?

Comment (MarkPotato777) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/PreCheckRuntimeFlowableTask.java:302]: Confirmed with @smallsheeeep , `PreCheckRuntimeFlowableTaskCopied` is not in use currently.

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**